src/components/table/table_header_cell.tsx

```diff
@@ -163,17 +163,18 @@ export const EuiTableHeaderCell: FunctionComponent<EuiTableHeaderCellProps> = ({
         <EuiInnerText>
           {(ref, innerText) => (
             <EuiI18n
-              token="euiTableHeaderCell.titleTextWithSort"
-              default="{innerText}; Sorted in {ariaSortValue} order"
-              values={{ innerText, ariaSortValue }}>
-              {(titleTextWithSort: string) => (
+              token="euiTableHeaderCell.titleTextWithDesc"
+              default="{innerText}; {description}"
+              values={{ innerText, description }}>
+              {(titleTextWithDesc: string) => (
                 <span
-                  title={
-                    isSorted ? titleTextWithSort : description || innerText
-                  }
+                  title={description ? titleTextWithDesc : innerText}
                   ref={ref}
                   className="euiTableCellContent__text">
                   {children}
+                  <EuiScreenReaderOnly>
+                    <span>{description}</span>
+                  </EuiScreenReaderOnly>
```

Comment:
Sorry, forgot about this in my example before, but this should be conditional so we don't render a bunch of empty `span` elements.

(Same goes for below on lines 235-237)

```suggestion
                  {description && <EuiScreenReaderOnly>
                    <span>{description}</span>
                  </EuiScreenReaderOnly>}
````